### ISSUE TITLE
Баллон с нулевым кПа

### ISSUE THREAD

**Issue by Mintt69** [TauCetiStation/TauCetiClassic#6289]
#### Подробное описание проблемы

Если выставить с новой менюшкой (менюшка иногда не работает когда нажимаешь на балон, я  например потыкал значения балона и он перестал открывать менюшку), давление мелкого балона в ноль, то не будет тратиться содержимое (кислород), а кукла не будет задыхаться
#### Что должно было произойти
Кислород с мелкого балона должен поступать в легкие куклы и тратиться
#### Что произошло на самом деле
Балон не тратится, кукла дышит нулём
#### Как повторить
Берем балон и с новой менюшкой выставляем значение 0 как ниже фотка и включаем дышим
#### Дополнительная информация:
Еще я попробовал покрутить в космосе ползунки, воздух забирало уже, а еще персонаж может разом всосать 300 кпа из балона, а еще перед тем как выйти в космос нужно иметь значение 0, чтобы не задыхаться, иначе если менять значение в космосе, то уже бесконечный воздух не работает, ну проверьте в общем
![d52ad30aff04e51e7e1e85683039d87905097b69 1](https://user-images.githubusercontent.com/36198983/97229928-2413de00-17ea-11eb-8a9b-0cf77ea8d9a1.png)

**Comment by Kortez90** [comment]
`![d52ad30aff04e51e7e1e85683039d87905097b69 1](https://user-images.githubusercontent.com/36198983/97229928-2413de00-17ea-11eb-8a9b-0cf77ea8d9a1.png)`
Попробуй использовать эту ссылку.

Потенциально.
#5990

**Comment by T6751** [comment]
https://github.com/TauCetiStation/TauCetiClassic/pull/6207

**Comment by AirBlack** [comment]
Кукла при нуле дышит воздухом из атмосферы, так как !0 = true, почему это работало нормально раньше не знаю, ща гляну

**Comment by AirBlack** [comment]
Код дыхания я не менял, а значит кукла и раньше могла всосать 300 кпа из балона

**Comment by Kortez90** [comment]
> кукла и раньше могла всосать 300 кпа из балона

Не могла.

![GIF 27 10 2020 20-09-31](https://user-images.githubusercontent.com/36198983/97336819-7dcee380-1890-11eb-857d-fdd9d2562a91.gif)

Даже если ты его не менял, этот участок могли зацепить "ползнуки".

**Comment by AirBlack** [comment]
Так, нашел, в чем дело. Походу раньше либо из-за неточности чисел с плавающей точкой либо хуй знает из-за чего distribute_pressure до конца в ноль не делался(или делался и работало все абсолютно так же), т.к. в 
https://github.com/TauCetiStation/TauCetiClassic/blob/b2353ae07c0ff5b38005a94e49835ec4493e875a/code/modules/atmospheric/xgm/xgm_gas_mixture.dm#L222
проверка на amount <= 0, при нулевом дистрибьюшоне будет эмаунт 0, код в life.dm посчитает, что что-то не так т.к. ему не вернулся объект микстуры воздуха и будет брать воздух из атмосферы.
То, как работают легкие и газовые микстуры, мой ПР не касается, поэтому поставлю минималку на 1, дабы такого не возникало.
На локалке только что проверил, как дышится с максимальным дистрибьюшоном, все точно так же, как и у тебя на гифке

**Comment by AirBlack** [comment]
> Даже если ты его не менял, этот участок могли зацепить "ползнуки".
Ты думаешь, что ты пишешь вообще? Посмотри ПР и посмотри, как было ДО и как стало ПОСЛЕ.

Вот то, как работает с моим ПРом
https://puu.sh/GHm8j/7bbbf3f7d9.gif
Сейчас загружу билд без ветки интерфейса и проверю как работало раньше

**Comment by AirBlack** [comment]
Проверил со старым интерфейсом, работает все абсолютно так же, как и с новым. Проблема не моего ПРа. При нулевом дистрибьюшоне так или иначе берется воздух из атмосферы, бесконечного воздуха нет. Working as intended.

**Comment by T6751** [comment]
Не является багом нового интерфейса

**Comment by TauKitty** [comment]
Эта проблема была упомянута в **Tau Ceti Station Forum**. Там могут быть соответствующие подробности:

https://forum.taucetistation.org/t/fludilka-chast-2/28659/16006

**Comment by Mercurialaste** [comment]
Не спасает, если находится в космосе. Т.е. работает только в местах, где кукла и так дышит тк она, дышит не из баллона. При этом, маска с включенным баллоном не защищает от вредных газов по типу слип газа.

**Comment by TEXH4Pb** [comment]
Бага никакого нет, Мерк прав
